src/background.vala

```diff
@@ -476,6 +476,12 @@ public class Background : Gtk.Fixed
         }
 
         set {
+            var pretty_value = value;
+            if (value == null) {
+                pretty_value = "<system default bg image>";
+            }
+            debug("Background change requested, changing to: %s", pretty_value);
```

Comment:
I'd do something like this instead:

```vala
       set {
            var pretty_value = "";

            if (value == null || value == "")
            {
                _current_background = system_background;
                pretty_value = "<system default bg image> ";
            } else
            {
                _current_background = value;
            }
            pretty_value += _current_background;

            debug("Background change requested, changing to: %s", pretty_value);
            reload ();
        }
```

This has the added benefit that we're not forgetting the empty case, which you missed...